src/display.jl

```diff
@@ -45,6 +45,51 @@ register_mime(x::AbstractVector{<:MIME}) = push!(ijulia_mime_types, Vector{Mime}
 register_jsonmime(x::Union{MIME, Vector{MIME}}) = push!(ijulia_jsonmime_types, x)
 register_jsonmime(x::AbstractVector{<:MIME}) = push!(ijulia_jsonmime_types, Vector{Mime}(x))
 
+
+# Check mime bundle dict key type and convert to string keys for JSON
+_format_mime_key(k::String) = k
+_format_mime_key(k::MIME) = string(k)
+_format_mime_key(k) = error("MIME bundle keys should be instances of String or MIME")
+_format_mimebundle(d::Dict{String}) = d
+_format_mimebundle(d::AbstractDict) = Dict(_format_mime_key(k) => v for (k, v) in pairs(d))
+
+"""
+    display_data(mime::Union{MIME, String}, data; metadata::Dict=Dict())
+    display_data(mimebundle::Dict; metadata::Dict=Dict())
+
+Publish raw data to be displayed on all Jupyter front ends.
```

Comment:
raw data is never a good term to use,
but here it seems particularly misleading.
that `data` argument is the output of  `limitstringmime` which means it has already going though a call to rerepresent it.
Which will have for example converted DataFrames to HTML tables, etc etc or anything else that has `show(io, mimetype, data)` defined on it for any of the registered mime types.

Reply:
How about something like "Publish rich display data to all Jupyter front ends"? I think the rest of the docstring is hopefully pretty clear on the expected format.

Reply:
bump?

Reply:
Could be.
I don't know enough about what this does to say for sure.
And I don't have merge rights here.